src/python/zquantum/core/interfaces/backend.py

```diff
@@ -120,14 +143,51 @@ def __init__(
         self.device_connectivity = device_connectivity
 
     @abstractmethod
+    def _get_wavefunction_from_native_circuit(
+        self, circuit: Circuit, initial_state=None
+    ) -> Wavefunction:
+        """Get wavefunction from circuit comprising only natively-supported operations.
+
+        Args:
+            circuit: circuit to simulate. Implementers of this function might assume
+              that this circuit comprises only natively-supported operations as decided
+              by self._is_supported predicate.
+            initial_state: amplitudes of the initial state. If `None`, implementers
+              should assume that the initial state is |0...0>, which typically does not
+              require any initialization.
+        Returns:
+            Wavefunction object encoding amplitudes of final state.
+        """
+
+    def is_natively_supported(self, operation: Operation) -> bool:
+        """Determine if given operation is natively supported by this Simulator.
+
+        This method will be used as a predicate in split_circuit function.
+        """
+        return isinstance(operation, GateOperation)
+
     def get_wavefunction(self, circuit: Circuit) -> Wavefunction:
         """Returns a wavefunction representing quantum state produced by a circuit
 
         Args:
             circuit: quantum circuit to be executed.
         """
-        self.number_of_circuits_run += 1
-        self.number_of_jobs_run += 1
+        state = np.zeros(2 ** circuit.n_qubits)
```

Comment:
Perhaps it should take the initial state as an optional argument? If `None` it would evaluate to |0...0>, but otherwise it would allow people to seed the simulation with whatever initial state they want?

Reply:
Great idea! Fixed in 1ce1140

Reply:
It lacks typing :)